R/tidy.R

```diff
@@ -55,8 +55,16 @@ tidy.recipe <- function(x, number = NA, id = NA, ...) {
   # If number is NA and ID is not, select the step with the corresponding
   # ID. Only a single ID is allowed, as this follows the convention for number
   num_oper <- length(x$steps)
-  if (num_oper == 0)
-    rlang::abort("No steps in recipe.")
+  if (num_oper == 0) {
+    return(
+      tibble(number = integer(),
+             operation = character(),
+             type = character(),
+             trained = logical(),
+             skip = logical(),
+             id = character())
+    )
+  }
```

Comment:
What happens if we just remove this branch entirely? It looks like our usage of `vapply()` should set us up pretty well for the 0-step case to "just work" without special handling? Maybe one or two calls to some base R function might need to be tweaked.

Reply:
Good eye! Appears to work without further modification